### ISSUE TITLE
Building on Linux fails

### ISSUE THREAD

**Issue by ghost** [rcgordon/Serious-Engine#71]
Compilation fails at 71% 

```
/usr/include/SDL2/SDL_assert.h:145:19: note: in definition of macro ‘SDL_enabled_assert’
         while ( !(condition) ) { \
                   ^~~~~~~~~
/home/ralf/Serious-Engine/Sources/./Engine/Base/Assert.h:104:26: note: in expansion of macro ‘SDL_assert’
     #define ASSERT(expr) SDL_assert(expr)
                          ^~~~~~~~~~
/home/ralf/Serious-Engine/Sources/./Engine/Templates/StaticArray.cpp:129:3: note: in expansion of macro ‘ASSERT’
   ASSERT(&arOriginal!=NULL);
   ^~~~~~
[ 51%] Linking CXX shared library Debug/libEntitiesMPD.so
[ 71%] Built target EntitiesMPD
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```
Tried both 32 and 64 bit

**Comment by stevenc99** [comment]
Hi,

The actual error message is not shown in the text you quoted.  Please scroll up to find the text "error 1" and "error:".  You probably have this problem:
> /build/Serious-Engine/Sources/Engine/Engine.cpp:131:0: error: unterminated #else
>  #if (defined USE_PORTABLE_C)  // rcg10072001

which could be fixed by either
https://github.com/rcgordon/Serious-Engine/pull/68/commits/a68ec8e5410a6809efac1f9cbbf316e5f43c77d5 or https://github.com/rcgordon/Serious-Engine/pull/69/commits/1984179657b4f38dcefdba43145bb23b4b7084df

**Comment by ghost** [comment]
Hi,

It appears to be a different issue.

[makelog.txt](https://github.com/rcgordon/Serious-Engine/files/868386/makelog.txt)

**Comment by stevenc99** [comment]
Hi RalfVB,

Your log file seems incomplete.  Perhaps you created it with " > makelog.txt" (which only captures stdout) but you would also need to append " 2>&1" if you want stderr (warnings/errors) to be recorded there too.

**Comment by ghost** [comment]
Hi,

You were right, it's indeed this error:

```
/build/Serious-Engine/Sources/Engine/Engine.cpp:131:0: error: unterminated #else
#if (defined USE_PORTABLE_C) // rcg10072001
```

I have tried both fixes separately, and together, doing a make clean inbetween, but I'm still unable to compile it

Here's the full log :[makelog.txt](https://github.com/rcgordon/Serious-Engine/files/868842/makelog.txt)

Thanks for your help

**Comment by stevenc99** [comment]
It may be easier to just clone ptitSeb's fork from https://github.com/ptitSeb/Serious-Engine which seems to fix all the compile issues for me.

**Comment by ghost** [comment]
Yes, that one worked for me as well, I've got both tfe and tse up and running. Thanks!

**Comment by doggodanubus** [comment]
This was for ssam-tfe, not tried it on ssam tse

I did get it to compile by gutting that part of Engine.cpp and cleaned up the other two errors but still crashed. Thanks for the link to ptitSeb but it still fails with 
 error: expected ‘;’ before ‘:’ token on WorldCollisionGrid.cpp

I gutted pitSeb's changes for arm and Pandora and it's compiled. It crashes when the game engine tries to display flames afaikt.  So a sludge of a step forward.

**Comment by ptitSeb** [comment]
You have a crash on linux with some flames? In TFE or TSE and on what level (on 32bits build right?)

**Comment by mad-student** [comment]
I also have some problems.
I started command Sources/build-linux64.sh and got this:
Will build with 'make -j4' ... please edit this script if incorrect.

    rm -rf cmake-build

    mkdir cmake-build

    cd cmake-build

    cmake -DCMAKE_BUILD_TYPE=Debug ..
    CMake Error: The source directory /Serious-Engine" does not appear to contain CMakeLists.txt.
    Specify --help for usage, or press the help button on the CMake GUI.
    Found CMakeLists.txt in "/Serious-Engine/Sources" and copy it to "~/Serious-Engine". After that restart command and got this:
    Will build with 'make -j4' ... please edit this script if incorrect.

    rm -rf cmake-build

    mkdir cmake-build

    cd cmake-build

    cmake -DCMAKE_BUILD_TYPE=Debug ..
    -- The C compiler identification is GNU 7.3.0
    -- The CXX compiler identification is unknown
    -- Check for working C compiler: /usr/bin/cc
    -- Check for working C compiler: /usr/bin/cc -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    CMake Error at CMakeLists.txt:2 (project):
    No CMAKE_CXX_COMPILER could be found.

    Tell CMake where to find the compiler by setting either the environment
    variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
    to the compiler, or to the compiler name if it is in the PATH.

-- Configuring incomplete, errors occurred!
See also "/Serious-Engine/cmake-build/CMakeFiles/CMakeOutput.log".
See also "/Serious-Engine/cmake-build/CMakeFiles/CMakeError.log".
There is this files that noted in "See also...".
CMakeError.log
CMakeOutput.log
Is there any possibility to fix that error?
P.S. Sorry for my English. This language isn't native for me.
I use Lubuntu 18.04 if this matter.

**Comment by nesper8** [comment]
[makelog.txt](https://github.com/rcgordon/Serious-Engine/files/5122282/makelog.txt)
Crashes on Arch x64

cmake version 3.18.2
GNU Make 4.3 Built for x86_64-pc-linux-gnu
gcc (GCC) 10.2.0

**Comment by ptitSeb** [comment]
What sources are you using @nesper8 ? because your log show
```
/home/nep/Projects/Serious-Engine/Sources/Engine/Engine.cpp:131: error: unterminated #else
  131 | #if (defined USE_PORTABLE_C)  // rcg10072001
```
~~But that line 131 is not like that in this repo.~~

Also, I assume you have a build issue, as your ticket is very minimal.

EDIT: Ah, I didn't noticed this was created in rcgordon repo. Try my fork: https://github.com/ptitSeb/Serious-Engine it should work.

**Comment by nesper8** [comment]
@ptitSeb , your version gets compiled out but returns a segmentation fault after the intro
which i have created a new issue on your fork

**Comment by ptitSeb** [comment]
I have see, but don't reproduce the issue. But the install process discribed in the README is a bit confusing and I need to clarify that out,e specially with the Free GoG version.

**Comment by coreybruce** [comment]
Having a similar issue when building on my Raspberry pi 4, I was able to get it this far tho, due to how long it is I had to upload it to pastebin.


https://pastebin.com/4s1jDgqx
